src/coreclr/src/jit/lsraarm64.cpp

```diff
@@ -1134,29 +1134,59 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree)
     {
         if (intrin.op2 != nullptr)
         {
-            if (isRMW)
+            bool op2DelayFree = isRMW;
+            bool op3DelayFree = isRMW;
+            bool op4DelayFree = isRMW;
```

Comment:
It would be good to add a comment above these, basically summarizing what you've said in the description of this PR.

Reply:
Thank you for review Carol! I pushed a commit that clarifies why the analysis was added.

I am also working on collecting data for a small example that demonstrates benefits of this optimization. I will post the disassembly on the thread before merging.